core/src/main/java/feast/core/job/JobUpdateTask.java

```diff
@@ -52,168 +49,130 @@
 @Getter
 public class JobUpdateTask implements Callable<Job> {
 
-  private final List<FeatureSetProto.FeatureSet> featureSets;
-  private final SourceProto.Source sourceSpec;
-  private final StoreProto.Store store;
+  private final List<FeatureSet> featureSets;
+  private final Source source;
+  private final Store store;
   private final Optional<Job> currentJob;
-  private JobManager jobManager;
-  private long jobUpdateTimeoutSeconds;
+  private final JobManager jobManager;
+  private final long jobUpdateTimeoutSeconds;
+  private final String runnerName;
 
   public JobUpdateTask(
-      List<FeatureSetProto.FeatureSet> featureSets,
-      SourceProto.Source sourceSpec,
-      StoreProto.Store store,
+      List<FeatureSet> featureSets,
+      Source source,
+      Store store,
       Optional<Job> currentJob,
       JobManager jobManager,
       long jobUpdateTimeoutSeconds) {
 
     this.featureSets = featureSets;
-    this.sourceSpec = sourceSpec;
+    this.source = source;
     this.store = store;
     this.currentJob = currentJob;
     this.jobManager = jobManager;
     this.jobUpdateTimeoutSeconds = jobUpdateTimeoutSeconds;
+    this.runnerName = jobManager.getRunnerType().toString();
```

Comment:
The `runnerName` as we have it above refers to the `runnerType` right? Should we deduplicate that from the configuration (we could do that at the configuration level as well)

```
feast:
  jobs:
    polling_interval_milliseconds: 60000
    job_update_timeout_seconds: 240

    active_runner: my_direct_runner

    runners:
      - name: my_direct_runner
        type: DirectRunner
        options: {}
```

https://github.com/gojek/feast/blame/ded7ca59a2bd5b40715af73fdaa4e4f19ad0b915/core/src/main/java/feast/core/config/FeastProperties.java#L95

Reply:
The idea with the configuration was to provide a more forward compatible configuration schema, but we could just as easily use type there instead of name.

Reply:
Yes, from your example config, `jobManager.getRunnerType().toString()` will be `"DirectRunner"` (whereas `jobManager.getRunnerType().name()` would be `"DIRECT"`).

In `JobUpdateTask` this value is being used only for log messages.